Revit_Core_Engine/Create/View/View3d.cs

```diff
@@ -0,0 +1,262 @@
+        {
+            //for information about a perspective boundingbox and its orientation see here:
+            //https://knowledge.autodesk.com/support/revit-products/learn-explore/caas/CloudHelp/cloudhelp/2014/ENU/Revit/files/GUID-A7FA8DBC-830E-482D-9B66-147399524442-htm.html
+
+            View3D result = null;
+
+            ViewFamilyType vft = Query.ViewFamilyType(document, ViewFamily.ThreeDimensional);
+
+            result = Autodesk.Revit.DB.View3D.CreatePerspective(document, vft.Id);
+            result.DetailLevel = viewDetailLevel;
+            
+            if (!string.IsNullOrEmpty(viewName))
+            {
+                try
+                {
+#if (REVIT2020 || REVIT2021)
+                    result.Name = viewName;
+#else
+                    result.ViewName = viewName;
+#endif
+                }
+                catch
+                {
+                    BH.Engine.Reflection.Compute.RecordWarning("There is already a view named '" + viewName + "'." + " It has been named '" + result.Name + "' instead.");
+                }
+            }
+            
+            if (viewTemplateId != null)
+            {
+                try
+                {
+                    result.ViewTemplateId = viewTemplateId;
+                }
+                catch (Exception)
+                {
+                    BH.Engine.Reflection.Compute.RecordWarning("Could not apply the View Template of Id " + viewTemplateId + "'." + ". Please check if it's a valid ElementId.");
+                }
+            }
+
+            if (viewOrientation3d != null)
+            {
+                try
+                {
+                    result.SetOrientation(viewOrientation3d);
+                }
+                catch (Exception)
+                {
+                    BH.Engine.Reflection.Compute.RecordWarning("Could not set the view's orientation in 3D due to unexpected values, please report this error.");
+                }
+            }
+
+            if (boundingBoxXyz != null)
+            {
+                try
+                {
+                    //transform to view coordinates previously picked points                    
+                    BoundingBoxXYZ bb = result.get_BoundingBox(result);
+                    Transform transform = bb.Transform;
+                    
+                    bb.Max = transform.Inverse.OfPoint(boundingBoxXyz.Max);
+                    bb.Min = transform.Inverse.OfPoint(boundingBoxXyz.Min);
+
+                    result.CropBox = bb;
+                    
+                    //changes the view's far clip to expected length of line based family (x1.5 offset) 
+                    Parameter viewFarClip = result.get_Parameter(BuiltInParameter.VIEWER_BOUND_OFFSET_FAR);
+                    viewFarClip.Set(boundingBoxXyz.Min.DistanceTo(boundingBoxXyz.Max) * 1.5);
```

Comment:
I think you should be able to use `SetParameter` here - as shown below, more or less:
```
    result.SetParameter(BuiltInParameter.VIEWER_BOUND_OFFSET_FAR, boundingBoxXyz.Min.DistanceTo(boundingBoxXyz.Max) * 1.5, false);
```